### ISSUE TITLE
PTv2: To `platform` or not to `platform`?

### ISSUE THREAD

**Issue by tjhorner** [tjhorner/gtfs-janitor#1]
As it stands right now, GTFS Janitor does not currently add `public_transport=platform` to nodes it creates or modifies. As noted on the [wiki](https://wiki.openstreetmap.org/wiki/Automated_edits/tjhorner-import#Tags):

> You may notice that `public_transport=platform` is omitted from this list. The platform is sometimes mapped as a separate way, so we don't necessarily want to add the tag to the `highway=bus_stop` node. For simplicity, determining whether or not to add `public_transport=platform` to the node is considered out of scope for this import. (Since the importer merges the tags, however, it will remain if it existed before, preventing that data from being erroneously erased.)

This decision was made simply because it was the easiest solution and could be resolved in the future. However, it's a bit controversial (understandably so) to omit the modern PTv2 tags. Let's discuss this problem here.

**Comment by tjhorner** [comment]
I'll start off by copy-pasting my response to a comment from someone in the OSM US Slack:

This was a decision made based on a discussion with a local mapper. She does a lot of micromapping of the bus stops in the area, so we actually have many separately-mapped `way[public_transport=platform]` with a nearby `node[highway=bus_stop]` (and both are part of a `relation[public_transport=stop_area]`). When mapping the stops, she includes `public_transport=platform` only on the platform way and not on the `highway=bus_stop` node, which I think is probably the most "correct" way of doing it (it makes sense not to duplicate them, IMO).We decided it was probably best to avoid duplication of the `public_transport=platform` elements, so there were a couple of approaches:

1.  Like you mentioned, check for a separately-mapped `public_transport=platform` way at the stop. If both the `highway=bus_stop` and `public_transport=platform` are part of the same `stop_area` relation, this is trivial to check. But it's [not guaranteed](https://overpass-turbo.eu/s/1P6F), so how should the check be done in this case? (See number 2)
2.  We can match using methods similar to how the stop itself was matched (match by ID -> name -> distance), but that could also falsely include, for example, a platform for a stop across the street. This would result in, using the previous example, the new node missing the `public_transport=platform` tag. (The disambiguation UI could ask the user about this but that would require many of changes to the tool, so I saved that as a future improvement.)
3.  Omit the `public_transport=platform` for newly-created nodes. This was the simplest approach and limited the scope of the import. The iD warning that this generates may actually be beneficial, because it defers the task of checking for a nearby `public_transport=platform` to a human mapper, so is more likely to be correct. (We can also correct this with automation in the future, or maybe a MapRoulette challenge or something.)

I guess the decision really depends on the weight we put on these factors:

1.  Mandatory existence of `public_transport=platform` on `highway=bus_stop` (i.e., PTv2 tagging)
2.  Duplication of multiple `public_transport=platform` elements

After writing all this out, I think the answer is probably something like "we should try not to duplicate `public_transport=platform`s, but if a small number of duplicates slip through then that's fine".Practically, that would probably look like a balance of the approaches described above. The tool could do a rudimentary check to see if the matched node is part of a `stop_area` relation that already has a `public_transport=platform` and leave the `public_transport` tag unchanged if that's the case (otherwise, add it). This means that we may inadvertently have a small number of duplicate `public_transport=platform` elements at stops with separately-mapped platform ways that *aren't* part of a relation, but we can clean these up manually with QA tools after the import. (I think this is probably an ideal solution since it does not require much extra effort, limits the number of errors the tool can make, and still allows us to place `public_transport=platform` on new nodes.Another potential solution: only add the `public_transport=platform` tag to newly-created nodes, not existing nodes that have been modified. This will reduce the overall number of PTv1-only nodes while still avoiding the erroneous addition of `public_transport=platform`. I just checked and there are actually [very few](https://overpass-turbo.eu/s/1P6N) `highway=bus_stop` nodes without `public_transport=platform` (and that aren't in a `stop_area` relation); those can likely be very easily manually reviewed since, at the time of writing, there are only 28 of them.

**Comment by tjhorner** [comment]
To resolve this point, I decided to do the following:

- The tool will only add `public_transport=platform` to _newly-created nodes_, not existing nodes that are being modified.
- I created a MapRoulette challenge to manually correct the various bus stops without PTv2 tagging (either by adding the tag or adding it to a `public_transport=stop_area` relation with the separately-mapped platform): https://mpr.lt/c/48887

It seems like a reasonable compromise that both avoids duplication but not create any more PTv1-only nodes. I updated the [wiki page](https://wiki.openstreetmap.org/wiki/Automated_edits/tjhorner-import#public_transport=platform_handling) to reflect this. If anyone has any objections to this, let me know. Otherwise I will probably close this issue in a day or two.
